src/dune/__main__.py

```diff
@@ -174,6 +174,26 @@ def load_all_modules_from_dir(plugin_dir):
                 else:
                     dune_sys.create_account(args.create_account[0])
 
+            elif args.system_newaccount is not None:
+                commands, WAS_REMAINDER_ARGS_USED = parse_optional(args.remainder)
```

Comment:
`parse_optional([])` returns `[],True` so maybe parse_optional has a defect that needs to be resolved?

Reply:
This scenario could happen only where "--" was passed without any arguments, right?
In this case `WAS_REMAINDER_ARGS_USED` should be True or False?
I don't think it makes any difference, but I went with your suggestion and now it will be "False" (while previously it was True).

Reply:
I did some white box testing and parse_optional was ALWAYS returning true, even if the return was an empty array (`[]`).